src/snowflake/snowpark/stored_procedure.py

```diff
@@ -59,6 +62,12 @@ def __init__(
         self._input_types = input_types
         self._execute_as = execute_as
 
+        api_call_source = api_call_source or "StoredProcedure.__init__"
+        session = session or snowflake.snowpark.session._get_active_session()
+        session._conn._telemetry_client.send_function_usage_telemetry(
+            api_call_source, TelemetryField.FUNC_CAT_CREATE.value
```

Comment:
This might not be correct. Creation of this class does not mean a stored proc is created.

The actually sp creation is in `StoredProcedureRegistration`. I'm wondering if we should add this `api_call_source` to `udf_utils#create_python_udf_or_sp`? This way we could get both udf and sp covered, though of course we need to pass the `api_call_source`.